### ISSUE TITLE
dynamic-blocks: An argument named "key" is not expected here

### ISSUE THREAD

**Issue by nextus** [hashicorp/packer#8967]
#### Overview of the Issue

According to the [documentation](https://packer.io/docs/configuration/from-1.5/expressions.html#dynamic-blocks), I have to use `key` and `value` in the `content` block but it doesn't work:

```
  on ./packer.pkr.hcl line 56, in source "amazon-ebs" "ebs":
  56:       key   = tag.key

An argument named "key" is not expected here.
```

It seems this default schema is applied: https://github.com/hashicorp/packer/blob/79f1fab8365b2c337d77c29aa20af09a05eedb99/hcl2template/types.variables.go#L295

I replaced in my configuration `key` to `name` as a workaround and everything starts working but this is implicit behavior.

#### Reproduction Steps

Run build with dynamic block from the documentation.

### Packer version

`1.5.5`

**Comment by sylviamoss** [comment]
Hey there, thanks for reaching out! We'll try to take a look for the next release.
Could you please provide us your simplified Packer hcl template?

**Comment by azr** [comment]
Hello there ! I first implemented it using the `key` key but then later changed it to `name` to remain consistent with [what Terraform has](https://www.terraform.io/docs/configuration/expressions.html#dynamic-blocks), updating the docs now.

**Comment by azr** [comment]
Ah scratch that, it's a little more complex. So when I implemented #8889 to allow to use tags/filters/subnetfilters key value maps I used the key "`name`" everywhere for consistency across Packer builders and Terraform I though; also it made implementation simpler.

But what Terraform providers do is using the naming convention of the upstream service. For example for the tags aws uses `key`/`value` ([docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html)) and in some cases it's `name`/`value` ([docs](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html)).

We'll soon work on fixing this to have this behaviour, note that this will be a breaking change.
